src/test/codegen/dst-vtable-align-nonzero.rs

```diff
@@ -42,4 +43,21 @@ pub fn does_not_eliminate_runtime_check_when_align_2(
     &x.dst
 }
 
+// CHECK-LABEL: @align_load_from_align_of_val
+#[no_mangle]
+pub fn align_load_from_align_of_val(x: &dyn Trait) -> usize {
+    // CHECK: {{%[0-9]+}} = load [[USIZE]], {{.+}} !range [[RANGE_META]]
+    core::mem::align_of_val(x)
+}
+
+// CHECK-LABEL: @align_load_from_vtable_align_intrinsic
+#[no_mangle]
+pub unsafe fn align_load_from_vtable_align_intrinsic(x: &dyn Trait) -> usize {
+    let (data, vtable): (*const (), *const ()) = core::mem::transmute(x);
+    // CHECK: {{%[0-9]+}} = load [[USIZE]], {{.+}} !range [[RANGE_META]]
+    let align = core::intrinsics::vtable_align(vtable);
+    // make this function unique so it doesn't get merged with the previous
```

Comment:
ymmv: you could use `-Z merge-functions=disabled` in the `compile-flags` instead, if you prefer.

(Like in https://github.com/rust-lang/rust/blob/f5418b09e84883c4de2e652a147ab9faff4eee29/src/test/codegen/vec-in-place.rs#L3 )